### PR TITLE
clickhouse: 20.5.2.7 -> 20.11.4.13

### PR DESCRIPTION
**Pull request by RaghavSood**
###### Motivation for this change

Follow up on #105224

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/clickhouse/default.nix [hunk 1]
```diff
@@ -46,7 +55,6 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [
     "-DENABLE_TESTS=OFF"
-    "-DUSE_INTERNAL_LLVM_LIBRARY=OFF"
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:49]: Why is this removed? We do not want to vendor LLVM.

**Comment by SuperSandro2000** [comment]
Build hangs at ``[1/4/6 built, 8 copied (48.8/48.9 MiB), 8.3 MiB DL] building clickhouse-20.11.4.13 (buildPhase): [0/2] Re-checking globbed directories...``

**Comment by RaghavSood** [comment]
Clickhouse takes forever to build - you're unlikely to be able to build it locally unless you have a beast of a machine

I'm using [AX41](https://www.hetzner.com/dedicated-rootserver/matrix-ax) machines, and it still takes 1 hour+

File: pkgs/servers/clickhouse/default.nix (hunk 1 above)
Comment (RaghavSood) [reply pkgs/servers/clickhouse/default.nix:49]: Was removed in the previous PR, I've added it back and updated the flags to keep up with upstream's requirements

Comment (lukegb) [reply pkgs/servers/clickhouse/default.nix:49]: It does take less time to compile when you're using the non-vendored LLVM though :^)

Comment (RaghavSood) [reply pkgs/servers/clickhouse/default.nix:49]: True, but I reckon most people compiling it from source won't care much about a one-time 10ish minute saving (at least on my hardware)

The majority of users will be grabbing it from cache.nixos.org

I'm open to making this configurable with a default to `OFF`

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 105968` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- clickhouse
</details>

**Review by lukegb — APPROVED** [review]
```

[nix-shell:~/.cache/nixpkgs-review/pr-105968]$ ./results/clickhouse/bin/clickhouse local --version
ClickHouse client version 20.11.4.13.

[nix-shell:~/.cache/nixpkgs-review/pr-105968]$ ./results/clickhouse/bin/clickhouse server --version
ClickHouse server version 20.11.4.13.
```

Yup, LG.
